Custom GraphQL schema in beta-3 is not working

<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/
-->

**Describe the bug**
We have just upgraded our project to Strapi version 3-beta, and the custom GraphQL schema that we have defined for user model in `/extensions/users-permissions/config/schema.graphql` is not working anymore, it's like this file is being ignored, although I haven't seen any change about this on the docs. It used to work in the latest alpha version `#alpha.26`.

**Steps to reproduce the behavior**
1. Write some custom GraphQL schema
2. Start the project
3. Use the custom query/mutation
4. See error

**Expected behavior**
The custom query/mutations that I've provided should work.

**Screenshots**
![Screen Shot 2019-06-09 at 16 45 23](https://user-images.githubusercontent.com/13555552/59158875-131fdb00-8ad6-11e9-9f93-8d1ac01104bd.png)

**Code snippets**
Here is the schema `/extensions/users-permissions/config/schema.graphql`:
```
module.exports = {
  query: `
    userSearch(_q: String!): [UsersPermissionsUser]
  `,
  resolver: {
    Query: {
      userSearch: {
        resolver: {
          plugin: 'users-permissions',
          handler: 'User.search'
        }
      },
    }
  }
};
```

Here is the query that will get sent to GraphQL endpoint:
```
query ($_q: String!) {
    userSearch(_q: $_q) {
        id
        first_name
        last_name
    }
}
```

Here is the error, when we request to that query:

```
{
    "errors": [
        {
            "message": "Cannot query field \"userSearch\" on type \"Query\". Did you mean \"users\"?",
            "locations": [{ "line": 2, "column": 3 }],
            "extensions": {
                "code": "GRAPHQL_VALIDATION_FAILED",
                "exception": {
                    "stacktrace": [
                        "GraphQLError: Cannot query field \"userSearch\" on type \"Query\". Did you mean \"users\"?",
                        "    at Object.Field (/project/node_modules/graphql/validation/rules/FieldsOnCorrectType.js:64:31)",
                        "    at Object.enter (/project/node_modules/graphql/language/visitor.js:332:29)",
                        "    at Object.enter (/project/node_modules/graphql/language/visitor.js:383:25)",
                        "    at visit (/project/node_modules/graphql/language/visitor.js:250:26)",
                        "    at Object.validate (/project/node_modules/graphql/validation/validate.js:63:22)",
                        "    at validate (/project/node_modules/apollo-server-core/dist/requestPipeline.js:211:32)",
                        "    at Object.<anonymous> (/project/node_modules/apollo-server-core/dist/requestPipeline.js:124:42)",
                        "    at Generator.next (<anonymous>)",
                        "    at fulfilled (/project/node_modules/apollo-server-core/dist/requestPipeline.js:4:58)",
                        "    at process._tickCallback (internal/process/next_tick.js:68:7)"
                    ]
                }
            }
        }
    ]
}
```

**System**
- Node.js version: v10.16.0
- NPM version: 6.9.0
- Strapi version: beta-3
- Database: PostgreSQL
- Operating system: MacOS

I have also asked this in StackOverflow: https://stackoverflow.com/q/56514188/5073175

@alexandrebodin like this?
```
"strapi-plugin-graphql": "file:../strapi/packages/strapi-plugin-graphql",
```

When I add the line above, I get this error:
```
[2019-06-11T11:08:56.203Z] debug ⛔️ Server wasn't able to start properly.
[2019-06-11T11:08:56.205Z] error Error: Cannot find module 'strapi/lib/load'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._resolveFilename (/Volumes/code/logickit-new/node_modules/module-alias/index.js:49:29)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
```

--- a/examples/getstarted/exports/graphql/schema.graphql
+++ b/examples/getstarted/exports/graphql/schema.graphql
@@ -216,6 +216,7 @@ type Query {
   test(id: ID!): MypluginTest
   tests(sort: String, limit: Int, start: Int, where: JSON): [MypluginTest]
   me: UsersPermissionsMe
+  userCustomRoute: String
 }
 
 input RoleInput {

--- a/examples/getstarted/extensions/users-permissions/config/schema.graphql.js
+++ b/examples/getstarted/extensions/users-permissions/config/schema.graphql.js
@@ -1,0 +1,15 @@
+module.exports = {
+  query: `
+    userCustomRoute: String
+  `,
+  resolver: {
+    Query: {
+      userCustomRoute: {
+        resolver: {
+          plugin: 'users-permissions',
+          handler: 'UsersPermissions.customRoute',
+        },
+      },
+    },
+  },
+};

--- a/packages/strapi-plugin-graphql/hooks/graphql/index.js
+++ b/packages/strapi-plugin-graphql/hooks/graphql/index.js
@@ -23,8 +23,11 @@ module.exports = strapi => {
       strapi.config.hook.load.after.push('graphql');
       // Load core utils.
 
-      const configs = await loadConfigs({ appPath, installedPlugins });
-      _.merge(strapi, configs);
+      const { api, plugins, extensions } = await loadConfigs({
+        appPath,
+        installedPlugins,
+      });
+      _.merge(strapi, { api, plugins });
 
       /*
        * Create a merge of all the GraphQL configuration.
@@ -37,11 +40,15 @@ module.exports = strapi => {
         _.get(strapi.plugins[key], 'config.schema.graphql', {})
       );
 
+      const extensionsSchemas = Object.keys(extensions || {}).map(key =>
+        _.get(extensions[key], 'config.schema.graphql', {})
+      );
+
       // save the final schema in the plugin's config
       _.set(
         strapi,
         ['plugins', 'graphql', 'config', '_schema', 'graphql'],
-        mergeSchemas([...apisSchemas, ...pluginsSchemas])
+        mergeSchemas([...apisSchemas, ...pluginsSchemas, ...extensionsSchemas])
       );
     },
 

--- a/packages/strapi-plugin-graphql/hooks/graphql/load-config.js
+++ b/packages/strapi-plugin-graphql/hooks/graphql/load-config.js
@@ -16,21 +16,26 @@ const loadPluginsGraphqlConfig = async installedPlugins => {
     );
     _.set(root, ['plugins', pluginName], result);
   }
+
   return root;
 };
 
 const loadLocalPluginsGraphqlConfig = async appPath =>
   loadUtils.loadFiles(appPath, 'plugins/**/config/*.graphql?(.js)');
 
+const loadExtensions = async appPath =>
+  loadUtils.loadFiles(appPath, 'extensions/**/config/*.graphql?(.js)');
+
 /**
  * Loads the graphql config files
  */
 module.exports = async ({ appPath, installedPlugins }) => {
-  const [apis, plugins, localPlugins] = await Promise.all([
+  const [apis, plugins, localPlugins, extensions] = await Promise.all([
     loadApisGraphqlConfig(appPath),
     loadPluginsGraphqlConfig(installedPlugins),
     loadLocalPluginsGraphqlConfig(appPath),
+    loadExtensions(appPath),
   ]);
 
-  return _.merge({}, apis, plugins, localPlugins);
+  return _.merge({}, apis, plugins, extensions, localPlugins);
 };

